### ISSUE TITLE
Validation layers 1.3.280 crash

### ISSUE THREAD

**Issue by corporateshark** [KhronosGroup/Vulkan-ValidationLayers#7746]
**Environment:**
 - OS: Windows 10
 - GPU and driver version: GeForce RTX 3070 Ti; NVIDIA 551.86
 - SDK or header version if building from repo: VulkanSDK 1.3.280
 - Options enabled (synchronization, best practices, etc.): GPU-assisted validation enabled

**Describe the Issue**

A crash in `Module::StaticData::StaticData()`:

```
// layers/state_tracher/shader_module.cpp:796
Module::StaticData::StaticData(const Module& module_state, StatelessData* stateless_data) {
      std::vector<uint32_t>::const_iterator it = module_state.words_.cbegin();
      it += 5;  // skip first 5 word of header
      while (it != module_state.words_.cend()) {
          Instruction insn(it);                                                       <--- crashed here because of an out-of-range iterator dereference
```

In this case, `module_state.words_` is empty so `it += 5` moves the iterator out of bounds.

**Reproducible example:**

* Run [`Tiny_Mesh`](https://github.com/corporateshark/lightweightvk/blob/master/samples/Tiny_Mesh.cpp).
* Press 'T'

**Expected behavior**

No crash.

**Comment by spencer-lunarg** [comment]
I'm trying to reproduce this locally (Linux Intel Mesa, ToT) don't see any crash

I do get 


```
vkCmdDrawIndexed():  Device address 0x6070008c8 access out of bounds.  Command buffer (0x5ac2fb889390). Draw Index 0. Pipeline (Pipeline: mesh)(0xf8c81f0000000143). Shader Module (Shader Module: main (frag))(0x85ef67000000013b). Shader Instruction Index = 167.  Stage = Fragment.  Fragment coord (x,y) = (270.5, 815.5).  Shader validation error occurred at line 55.Unable to find SPIR-V OpSource.
```

@corporateshark do you have the SPIR-V file that is breaking this, otherwise I will need to wait until Monday to get a NVIDIA Windows machine to reproduce

**Comment by spencer-lunarg** [comment]
This is also a suspicious crash because this is our most basic "parse the SPIR-V file" logic in normal core checks... if this is failing, something is going very wrong and not sure how we are getting here with a blank `module_state.words_`

**Comment by corporateshark** [comment]
@spencer-lunarg the crash is stable (happens every time I press `T` in the app). Here's the callstack:

![image](https://github.com/KhronosGroup/Vulkan-ValidationLayers/assets/2510143/427fd3bb-eddc-4aa7-a1d8-58af45b37c11)

It looks like `UtilGenerateSourceMessages()` receives an empty span `vvl::span<const uint32_t> pgm`.

Besides that, I believe doing that unconditional `it +=5` and not actually checking the results is a bit overoptimistic.

> I do get

That validation error is another story. It appeared first in VulkanSDK 1.3.275.

> do you have the SPIR-V file that is breaking this

Here're vertex/fragment shaders [dump_spirv.zip](https://github.com/KhronosGroup/Vulkan-ValidationLayers/files/14733563/dump_spirv.zip).

**Comment by spencer-lunarg** [comment]
The stack trace was helpful, I see the root cause!

> Besides that, I believe doing that unconditional it +=5 and not actually checking the results is a bit overoptimistic.

I agree its a bit overoptimistic, but I think the most I would do is assert because so many places around the code assume that if it made it that far, it is valid SPIR-V 

... The "real" issue is I forgot GPU-AV tries to extract `OpLine` information and no one is checking for valid SPIR-V along the way

Will get a good fix in for this tomorrow

**Comment by spencer-lunarg** [comment]
@corporateshark I pushed a quick fix to unblock this (I will do the proper cleanup once the other large PR for GPU-AV lands)
